great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2765,18 +2772,21 @@ def _get_batch_list(
             path=path,
             batch_filter_parameters=batch_filter_parameters,
             batch_spec_passthrough=batch_spec_passthrough,
+            batch_request_options=batch_request_options,
             **kwargs,
         )
-        datasource_name = batch_request.datasource_name
-        if datasource_name in self.datasources:
-            datasource: Datasource = cast(Datasource, self.datasources[datasource_name])
-        else:
+        datasource_name = result.datasource_name
```

Comment:
The casting that used to be here was misleading, since many datasource types could be returned. Now when you use an IDE to find the definition of `get_batch_list_from_batch_request` it tells you to look in multiple types, which is correct given the Union key value of the underlying `self.datasources` dict. It really upset mypy though (see ignore below). Please let me know if there is a better way

Reply:
@donaldheppner The `cast` did make sense before FDS, because all `Datasource` subclasses extended the parent top-level `Datasource` class, and, because the IDE does not "sense" polymorphism, it would display a warning, which the `cast` alleviated.  On a separate note, could we please revert `result` back into `batch_request` (or some name that indicates that this is a "sanitized" version of the `batch_request`)?  Thanks.

Reply:
I intentionally changed it to `result` to disambiguate it from the `batch_request` parameter. It was very confusing before as we were overwriting the parameter value